### DISCUSSION TITLE
How to filter array data using multiple condition?

### DISCUSSION THREAD

**Discussion by emaildipen** [pydata/xarray#6278] [Q&A] [answered]
I tried this but it didn't work, 

NT['DN'] = NTL['DN'].where(NT['Cloud_Mask']==50&58)

It only filters the array values with NTL['Cloud_Mask']==50, in fact, I want to filter it using NT['Cloud_Mask'] that have values 50 and 58.

**Comment by keewis** [comment, marked as answer]
the operator precedence causes `python` to interpret your expression as:
```python
NT['Cloud_Mask'] == (50 & 58)
```
which means that it will first compute the bitwise-and of `50` and `58` (the result is `50`), after which it will compare that to your array. To fix that, we need to introduce parens:
```python
(NT['Cloud_Mask'] == 50) & (NT['Cloud_Mask'] == 58)
```

However, you probably want to use [`isin`](https://docs.xarray.dev/en/latest/generated/xarray.DataArray.isin.html) instead:
```python
NT['Cloud_Mask'].isin([50, 58])
```
